### ISSUE TITLE
Escape instruction is not handled

### ISSUE THREAD

**Issue by okainov** [hadolint/hadolint#371]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Steps to reproduce the behavior
Try to lint Windows dockerfile:

```Dockerfile
#escape = `

FROM mcr.microsoft.com/dotnet/framework/runtime:4.8

ENV something=something `
        somethingelse=somethingelse
```

hadolint complains about "unexpected newline", seems it's not able to handle `escape` and \` symbol as newline indicator

**Comment by paulcalabro** [comment]
I think I'm experiencing this error as well:

```
/dev/stdin:73:74 unexpected 'a' expecting a new line followed by the next instruction or at least one space
```

```
...
LABEL ofelia.job-exec.some_job.command="\
  python -c \"import jobs;\
...
```

...it complains about the _i_ in **import**.

**Comment by mloskot** [comment]
I can also confirm the `escape=` directive is ignored and it triggers the ShellCheck

```Dockerfile
# escape=`
FROM mcr.microsoft.com/windoows/nanoserver:20H2 AS build
RUN move .\x64\Software .\Software
RUN rmdir /S /Q .\x64
```

```
Dockerfile:13 SC1001 info: This \S will be a regular 'S' in this context.
```

**Comment by lorenzo** [comment]
Yeah, we currently have no support for the escape directive in the parser
